### PR TITLE
Add typescript typings.

### PR DESCRIPTION
**Pull request by techmunk**

### CODE REVIEW

**Comment by techmunk** [comment]
TL;DR Yes, I'm currently using it in an app. I've been running with this code for about a week now. As it's just a declaration file it doesn't impact any functionality, just lets the typescript compiler compile to JS without errors/warnings.

I'm currently using it in a typescript app as follows.
```
import GracefulExit from 'express-graceful-exit'
...
GracefulExit.init(server)
app.use(GracefulExit.middleware(app))
...
GracefulExit.gracefulExitHandler(app, server, GracefulExit.gracefulExitHandler(app, server, {
    log: true,
    logger: (message: string) => log.info(message),
    callback: (code: number) => {
      log.mark(code === 0 ? 'Server has gracefully closed all open connections and shutdown' : 'Server was forced to close connections and has abruptly shutdown.')
      // Perform the shutdown callback on the next tick to allow time for our final log message to get sent.
      !hotreload && process.nextTick(() => shutdown(() => process.exit(code)))
    },
    performLastRequest: true,
    errorDuringExit: true,
    getRejectionError: () => {
      return CreateHttpError(503, 'Server is in the process of shutting down.')
    },
    exitProcess: false,
    // Docker has a hard timeout of 10 seconds. Make sure we're under that.
    suicideTimeout: hotreload ? 0 : 8,
    force: true
  }))
```

This pull request provides the typescript type information about the express graceful exit API so we get the typescript benefits of static type checking while still using a pure JS library. This doesn't actually affect how the JS code works, it just provides type hinting to the typescript compiler and some intellisense support in IDE's.

I have not actually used the three definitions for
```
function disconnectSocketIOClients(): void
function hardExitHandler(): void
function handleFinalRequests(req: Request, res: Response, next: NextFunction): void
```

They're on the export so I included them, but they some more like internal module calls. I can see a use case though where someone might want more control which I why I assumed they were exposed.

**Comment by kevynb** [comment]
@ivolucien I would love to see that merged because I'm also using `express-graceful-exit` in one of my Typescript projects.

Is there anything I can do to help speed up that process?

I agree with @techmunk when he says that the typings have no impact on the Javascript code. If you're not using Typescript you won't see a difference.

I would only suggest to add `"types": "index.d.ts",` to the `package.json` to wire types correctly for any typescript config.

**Comment by ivolucien** [comment]
@techmunk, @kevynb - taking a look now, and that you for being polite when I so rudely dropped the ball on this PR.

**Comment by ivolucien** [comment]
@techmunk, @kevynb - I've merged the changes to master, please take a look and let me know if the ts file and package.json changes work without issue.  Once it's 100% I'll add a note in the docs and bump the minor version.

**Comment by kevynb** [comment]
Hey @ivolucien! Sorry for the delay. On my side I've got two errors:

The first one: it seems a `declare` is missing before `namespace GracefulExit` because my typescript compiler complains about it. The second one is that the `socketio` property is missing from the `Configuration` Interface.

I cannot compile because of that.

I would suggest the following edits:

```typescript
> declare namespace GracefulExit {
  
 > interface DisconnectableSocket {
 >   disconnect(): void
 > }
  
  interface Configuration {
    errorDuringExit?: boolean
    performLastRequest?: boolean
    callback?: (code: number) => void
    log?: boolean
    logger?: (message: string) => void
    getRejectionError?: () => Error
    suicideTimeout?: number
    exitProcess?: boolean
    exitDelay?: number
    force?: boolean
   > socketio: {
   >   sockets: {
   >     sockets?: Record<any, DisconnectableSocket> | Array<DisconnectableSocket>
   >     clients?(): Array<DisconnectableSocket>
   >   }
   >   close?(): void
   > }
  }
...
```
